### PR TITLE
feat: Campsite conferencing app

### PR DESCRIPTION
**Pull request by nholden**
## What does this PR do?

Your friends at 🏕️ [Campsite](https://www.campsite.co/) are excited to propose a Campsite conferencing app on Cal.com. This PR implements that app. When installed, users will be able to set Campsite as the location for event types, and each new booking will be assigned a fresh Campsite call link just for that booking.

The app authenticates with the Campsite API by creating and storing an OAuth access token. It retrieves Campsite call links to attach to new bookings by using the access token to make a request to the Campsite API.

I followed [How to build an app](https://cal.com/docs/how-to-guides/app-store-and-integrations/how-to-build-an-app#steps-in-creating-an-app) to get started, and I referenced https://github.com/calcom/cal.com/pull/13929 as prior art for generating dynamic call links.

⚠️ In development, when I set Campsite as the default conferencing app for all event types, new bookings still defaulted to Cal Video. It was only after I manually changed the location for a specific event type that new bookings began to use Campsite. Let me know if you notice something I've done incorrectly here that might have caused that! 🙏 

https://github.com/calcom/cal.com/assets/7942714/6b8f8997-76be-46cf-9eb3-2ebf376fb9f5

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox. N/A
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works. (I did not add automated tests for this app. Do you have any suggestions as a starting point?)

## How should this be tested?

Please contact nick@campsite.co for the `client_id` and `client_secret` that you'll need to set to test this change. If you plan to test this with an OAuth callback URL other than `https://app.cal.com/api/integrations/campsite/callback`, let me know so that I can configure Campsite to accept those requests.

### CODE REVIEW

File: packages/app-store/campsite/api/callback.ts
Comment (SomayChauhan) [inline packages/app-store/campsite/api/callback.ts:67]: could you add returnTo here similar to what we have in zoom callback 

https://github.com/calcom/cal.com/blob/92845d3adc702e2cf01a19519bac40af2f3a74df/packages/app-store/zoomvideo/api/callback.ts#L81

**Review by ibex088 — CHANGES_REQUESTED** [review]
Hey @nholden!

Super psyched to see the campsite integration with Cal. Thanks for the PR! I left a small comment, but otherwise, it looks great!

File: packages/app-store/campsite/api/callback.ts
Comment (SomayChauhan) [reply packages/app-store/campsite/api/callback.ts:67]: this would allow campsite to be redirected to the app install flow
Screenshot 2024-07-10 at 9 22 44 PM

Comment (nholden) [reply packages/app-store/campsite/api/callback.ts]: Oh, nice! That's way better. Thanks for the suggestion. Updated in https://github.com/calcom/cal.com/pull/15713/commits/922265f63436d6b9fe6412b90ff1e111f50bb7d3.

**Review by ibex088 — APPROVED** [review]
LGTM!!
will wait for another approval!!
for anyone reviewing if you want `client_id` and `client_secret` you can reach out to me

**Comment by nholden** [comment]
Thanks for your help and for merging this PR! Excited to try this out in production. Do you have a rough timeline for enabling the app, or is there anything I can do to help?

**Comment by SomayChauhan** [comment]
Hi @nholden,

It should be available in production with our next release, which I believe is scheduled for next Tuesday. 
if I’m not mistaken, the credentials for the callback endpoint you sent (https://app.cal.com/api/integrations/campsite/callback), should work, right?

**Comment by nholden** [comment]
Thanks, @SomayChauhan! Sounds great.

> the credentials for the callback endpoint you sent (https://app.cal.com/api/integrations/campsite/callback), should work, right?

Yes!

**Pull request merged**
